### PR TITLE
Initial implementation of prim's algorithm

### PR DESCRIPTION
**Pull request by dkgoutham**
Implementation of prim's algorithm

### CODE REVIEW

File: src/AI-Algorithms-Graph/AIPrim.class.st [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	firstNode := nodes first.
+	firstNode distance: 0.
+
+	numNodes := nodes size.
+
+	numNodes timesRepeat: [ 
+			curNode := self minNode.
+			edges do: [ :edge |
+					curEdge := edge asTuple.
+					(curEdge first == curNode) ifTrue: [
```
Comment (dkgoutham) [inline src/AI-Algorithms-Graph/AIPrim.class.st:47]: This is the line causing the trouble, verified using the debugger. even though the nodes are equal, it isn't recognising them as equal nodes.

**Comment by dkgoutham** [comment]
> 
Screenshot 2023-04-04 at 2 14 13 AM

Screenshot 2023-04-04 at 2 15 04 AM
2" alt="Screenshot 2023-04-04 at 2 14 43 AM" src="https://user-images.githubusercontent.com/75137173/229623735-732737b8-5ff4-49c8-a780-5b71fe420d38.png">

**Comment by jordanmontt** [comment]
Hello, the Prim's algo code seems good! But, you also changed the code of the Dijkstra algo. This is because someone else committed to the library and you didn't pull the changes. So, can you please remove the changes to Dijkstra and to the PAthDistanceNode. Also can you add some tests to test this algo? The tests should tests from basic graph to edge cases
